include/swift/SIL/SILInstruction.h

```diff
@@ -343,8 +373,150 @@ class UnaryInstructionBase : public BASE {
   typename std::enable_if<has_result<X>::value, SILType>::type
   getType() const { return ValueBase::getType(); }
 
-  ArrayRef<Operand> getAllOperands() const { return Operands.asArray(); }
-  MutableArrayRef<Operand> getAllOperands() { return Operands.asArray(); }
+  ArrayRef<Operand> getAllOperands() const {
+    return {&Operands.asArray()[0],
+            static_cast<size_t>(1 + (hasOpenedArchetypesOperands() ? 1 : 0))};
```

Comment:
It's not obvious to me that a unary instruction can only have one opened archetype operand. Could you explain that somewhere? Then could you either use getNumOpenedArchetypesOperands() or assert that there's only one archetype operand?